### ISSUE TITLE
Resend verification mail when an unverified account try to log in

### ISSUE THREAD

**Issue by Gakamine** [graphql-devise/graphql_devise#40]
Hello, I'm requesting this enhancement in case my SMTP server is down or my user not receiving the mail to verify his/her email for some reasons.
What I suggest is to resend a confirmation instruction mail when a user try to log in with an unconfirmed account.
Thanks !

**Comment by mcelicalderon** [comment]
Hey @Gakamine. Right now #35 is soon to be merged and will add a resend confirmation instructions mutation. But this is not a bad idea, WDYT, @00dav00 ? We can probably make it configurable so the default behavior remains the same. One important change would be that the login mutation would need to add an optional `confirm_redirect_url` as the sign_up mutation does. Then we can check the default value configured in the initializer if no value is provided in the mutation, and raise an error if none is provided and the gem is configured to always send the confirmation email on a login attempt for a non confirmed account

**Comment by mcelicalderon** [comment]
I discussed with @00dav00 and we are not sure this is something the gem should implement as looks more like something your specific app might need to implement. A simple way to do it right now would be to override the default mutation and do something like this
```ruby
def resolve(email:, password:)
  resource = resource_class.find_by(email: email)
  
  resource.send_confirmation_instructions(
    redirect_url: redirect_url,
    template_path: ['graphql_devise/mailer']
  ) && raise_user_error(I18n.t('devise_token_auth.sessions.not_confirmed', email: resource.email)) if resource.present? && !resource.confirmed?

  super(email: email, password: password)
end
```
After writing that piece of code I see how it might make more sense to make that part of the gem, so we'll keep this issue open until we decide what to do, implementing all this possible application logics in the gem would also be a mistake.

**Comment by mcelicalderon** [comment]
Also, #35 is merged and will be released on `v0.7.0` tomorrow. Usually apps add a resend confirmation link in the event the email never arrives.

**Comment by mcelicalderon** [comment]
Closing this one as this is not something generic the gem should implement, but something specific to some applications. Current version of the gem does allow you to implement this flow in your app as demonstrated on the code snippet above. 

Also, I don't think I've ever used an app that automatically resends the confirmation email upon trying to login to an unconfirmed account. A more common flow would be to display a screen explaining the account is not confirmed and has a `send confirmation email` button for the user as the gem does have a mutation to resend the confirmation email.
